src/lib/integration_test_abstract_engine/abstract_network.ml

```diff
@@ -0,0 +1,1292 @@
+                  "File already exists for precomputed block with state hash %s"
+                  state_hash
+            | _ ->
+                [%log info]
+                  "Dumping precomputed block with state hash %s to file %s"
+                  state_hash filename ;
+                Out_channel.with_file filename ~f:(fun out_ch ->
+                    Out_channel.output_string out_ch block ) )
+      in
+      Malleable_error.return ()
+    with Failure err -> Malleable_error.hard_error_string err
+end
+
+type t =
+  { constants : Test_config.constants
+  ; testnet_log_filter : string
+  ; genesis_keypairs : Network_keypair.t Core.String.Map.t
+        (* below values are given by CI *)
+  ; seeds : Node.t Core.String.Map.t
+  ; block_producers : Node.t Core.String.Map.t
+  ; snark_coordinators : Node.t Core.String.Map.t
+  ; snark_workers : Node.t Core.String.Map.t
+  ; archive_nodes : Node.t Core.String.Map.t
+  ; network_id : Network_id.t
+  }
+
+let id { network_id; _ } = network_id
+
+let constants { constants; _ } = constants
+
+let constraint_constants { constants; _ } = constants.constraints
+
+let genesis_constants { constants; _ } = constants.genesis
+
+let seeds { seeds; _ } = seeds
+
+let block_producers { block_producers; _ } = block_producers
+
+let snark_coordinators { snark_coordinators; _ } = snark_coordinators
+
+let snark_workers { snark_workers; _ } = snark_workers
+
+let archive_nodes { archive_nodes; _ } = archive_nodes
+
+(* all_nodes returns all *actual* mina nodes; note that a snark_worker is a pod within the network but not technically a mina node, therefore not included here.  snark coordinators on the other hand ARE mina nodes *)
+let all_nodes { seeds; block_producers; snark_coordinators; archive_nodes; _ } =
+  List.concat
+    [ Core.String.Map.to_alist seeds
+    ; Core.String.Map.to_alist block_producers
+    ; Core.String.Map.to_alist snark_coordinators
+    ; Core.String.Map.to_alist archive_nodes
+    ]
+  |> Core.String.Map.of_alist_exn
+
+let all_pods t =
+  List.concat
+    [ Core.String.Map.to_alist t.seeds
+    ; Core.String.Map.to_alist t.block_producers
+    ; Core.String.Map.to_alist t.snark_coordinators
+    ; Core.String.Map.to_alist t.snark_workers
+    ; Core.String.Map.to_alist t.archive_nodes
+    ]
+  |> Core.String.Map.of_alist_exn
+
+(* all_non_seed_pods returns everything in the network except seed nodes *)
+let all_non_seed_pods t =
+  List.concat
+    [ Core.String.Map.to_alist t.block_producers
+    ; Core.String.Map.to_alist t.snark_coordinators
+    ; Core.String.Map.to_alist t.snark_workers
+    ; Core.String.Map.to_alist t.archive_nodes
+    ]
+  |> Core.String.Map.of_alist_exn
+
+let genesis_keypairs { genesis_keypairs; _ } = genesis_keypairs
+
+let all_node_id t =
+  let pods = all_pods t |> Core.Map.to_alist in
+  List.fold pods ~init:[] ~f:(fun acc (_, node) -> node.node_id :: acc)
+
+(* TODO: expand or remove *)
+let initialize_infra ~logger network =
```

Comment:
We can't remove this as it's in the intf.ml file and is called in the `test_executive` itself. To minimise changes to the `cloud` engine we should keep the function as is below. In the GCP case, the function looks like it's just testing all the pods were deployed with nodes in them. Given that we've abstracted away the infrastructure from the test I assume this will now  be done in the network runner (e.g. minimina or AWS), so we don't need anything other than a unit returning function, but we will need some process like this in the network runner that basically checks the network is ready to start the nodes.